packages/react/src/components/DateTimePicker/DateTimePicker.jsx

```diff
@@ -682,6 +686,7 @@ const DateTimePicker = ({
           className={classnames(`${iotPrefix}--date-time-picker__menu`, {
             [`${iotPrefix}--date-time-picker__menu-expanded`]: isExpanded,
           })}
+          style={{ '--zIndex': zIndex }}
```

Comment:
@jessieyan, additionally if there is not z index we are rendering a style attribute and changing the zIndex custom property to undefined.

Reply:
@davidicus  I've updated it. Please have another review.